### PR TITLE
New language/theme selection

### PR DESCRIPTION
**Pull request by Aloso**
Puts this in the top right corner:

tr

And in the footer:

footer-bright

footer

This makes the language selection more accessible by showing it at the top of the start page. The theme can now be changed on every page in the footer. The controls in the top right corner are not visible on small devices, but the ones in the footer are always visible. I hope the new UI is more intuitive than the old one.

### CODE REVIEW

**Comment by 94726** [comment]
I love the design, looks great. But I'm not sure about the language switch. I don't know whether anything is planned, but if we end up adding more languages, it would probably end up needing to be changed. 

Going to review the code sometime today.

**Comment by Aloso** [comment]
Adding 1 or 2 more languages wouldn't be an issue with this design. How many other translations would you be able to contribute? More importantly, which other languages do you think are needed? I assumed that there aren't any students who speak neither German nor English.

**Comment by 94726** [comment]
> How many other translations would you be able to contribute

Just roasting me here lol. I could try my best at polish though, but nearly no one speaks that anyways.

> I assumed that there aren't any students who speak neither German nor English.

Yeah you're probably right.

File: src/components/SelectSlider.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+import styles from "./SelectSlider.module.scss";
+
+interface SelectProps<T extends string> {
+  name: string;
+  options: Option<T>[];
+  value: T;
+  onChange(value: T): void;
+}
+
+type Option<T extends string> = T | { name: T; node: ReactNode };
```
Comment (94726) [inline src/components/SelectSlider.tsx:12]: I took quite a bit of time, trying to understand the props.
Some little documentation about what all of these do in conjunction with each other like ->
```ts
value: T; // String identifying the selected element
```
maybe what role  `T` or `name` of `type Option` play in that (might choose a better name than `name` like `value` 🤷).

File: src/sections/Footer.tsx [hunk 2]
```diff
@@ -1,19 +1,17 @@
...
+  const dispatch = useDispatch();
   const { t, i18n } = useTranslation();
-  const handleLanguageChange = () => {
-    if (i18n.language === "en-US") {
-      i18n.changeLanguage("de-DE");
-    } else {
-      i18n.changeLanguage("en-US");
-    }
-  };
+  const { theme } = useTheme();
```
Comment (94726) [inline src/sections/Footer.tsx:14]: remove the now unnecessary stuff

**Review by 94726 — CHANGES_REQUESTED** [review]

**Comment by LeoTuet** [comment]
> I love the design, looks great. But I'm not sure about the language switch. I don't know whether anything is planned, but if we end up adding more languages, it would probably end up needing to be changed.
> 
> Going to review the code sometime today.

we will stick to only english and german

**Review by LeoTuet — APPROVED** [review]

**Review by 94726 — APPROVED** [review]
👍

**Pull request merged**